src/forms/cashflow.js

```diff
@@ -42,68 +43,95 @@ const CashFlowContainer = function ({ children, label, validRow, message }) {
-        displayValidator: hasOnlyNonNegNumberChars,
-        storeValidator:   isNonNegNumber,
-        format:           toMoneyStr,
-      };
-
-  return (
-    <CashFlowContainer
-      label={ children }
-      validRow={ true }
-      message={ null }>
-      <ManagedNumberField
-        { ...baseProps }
-        value     = { baseVal / (4 + 1 / 3) }
-        otherData = {{ interval: 'weekly' }} />
-      <ManagedNumberField
-        { ...baseProps }
-        value     = { baseVal }
-        otherData = {{ interval: 'monthly' }} />
-      <ManagedNumberField
-        { ...baseProps }
-        value     = { baseVal * 12 }
-        otherData = {{ interval: 'yearly' }} />
-    </CashFlowContainer>
-  );
-
-};  // End CashFlowInputsRow{} Component
+class CashFlowInputsRow extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { message:'' }
+    // Bind the validator to set message
+    this.cashFlowStoreValidator = this.cashFlowStoreValidator.bind(this);
+  }
+
+  // Special store validator that handles maximums and sets error message
+  cashFlowStoreValidator(max) {
+    return function (str) {
+      if(!isNonNegNumber) {
+        this.setState({ message: 'Invalid number format' });
+        return false;
+      }
+      else if(parseFloat(str) > max) {
+        this.setState({ message: 'The input number exceeds the maximum of $999,999.99/yr' });
+        return false;
+      }
+      this.setState({ message: '' });
+      this.state.message = ''
+      return true;
+    }.bind(this); // bind this too
+  }
+
+  render() {
+    var { generic, timeState, updateClientValue, children } = this.props;
+
+    var updateClient = function (evnt, inputProps, data) {
+      var monthly = toMonthlyAmount[ data.interval ](evnt, inputProps.value),
+          obj     = { name: generic, value: monthly };
+      updateClientValue(evnt, obj);
+    };
+  
+    
+  
+    /** baseVal
+     * Get the time ('future' or 'current') monthly value unless there is
+     *     none, in which case, get the 'current' monthly cash flow value
+     *     (to prefill future values with 'current' ones if needed).
+     *
+     * @var
+     *
+     * @todo Add some kind of UI indication when it's the same as the 'current'
+     *     value. What if some of the row's values are the same and some are
+     *     different?
+     */
+    var baseVal   = timeState[ generic ],
+        baseProps = {
+          name:             generic,
+          className:        'cashflow-column',
+          store:            updateClient,
+          displayValidator: hasOnlyNonNegNumberChars,
+          format:           toMoneyStr,
+        };
+
+    var cashFlowStoreValidator = this.cashFlowStoreValidator;
+  
+    return (
+      <CashFlowContainer
+        label={ children }
+        validRow={ !this.state.message }
+        message={ this.state.message }>
+        <ManagedNumberField storeValidator={cashFlowStoreValidator(19230.76)}
```

Comment:
Instead of hard-coding the limits in this and the following lines (magic numbers), it would be better to assign the number that is the source of the others (999999.99) to a descriptively-named variable, and then obtain the others through arithmetic. That way, if the input constraints change, we only have to change one number and it's easier to understand how to do that.